v1/hash_test.go

```diff
@@ -58,7 +60,10 @@ func TestBadHashes(t *testing.T) {
 }
 
 func TestSHA256(t *testing.T) {
-	h := SHA256("asdf")
+	h, err := SHA256(ioutil.NopCloser(bytes.NewBufferString("asdf")))
```

Comment:
https://golang.org/pkg/strings/#NewReader ?

Reply:
Doesn't avoid the `NopCloser`.  I like `bytes.Buffer` :)